starter-code/src/functions-and-arrays.js

```diff
@@ -1,93 +1,473 @@
 // Iteration #1: Find the maximum
 
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  } else if (num1 < num2) {
+    return num2;
+  } else {
+    return num1;
+  }
+}
+
 // Iteration #2: Find longest word
-const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+const words = [
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
+];
+
+function findLongestWord(array) {
+  let nameMoreLong = "";
+  let count = 0;
+
+  if (array.length < 1) {
+    return null;
+  }
+  // else if (array.length == 1) {
+  //   return array[0]
+  // }
+  // else {
+  for (i = 0; i < array.length; i++) {
+    if (array[i].length > count) {
+      count = array[i].length;
+      nameMoreLong = array[i];
+    }
+  }
+  return nameMoreLong;
+  // }
+}
 
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(array) {
+  let sum = 0;
+  for (i = 0; i < array.length; i++) {
+    sum += array[i];
+  }
+  return sum;
+}
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers(array) {
+  if (array < 1) {
+    return null;
+  }
+  let sum = 0;
+  for (i = 0; i < array.length; i++) {
+    sum += array[i];
+  }
+  return sum / array.length;
+}
+
 // Level 2: Array of strings
 const wordsArr = [
-  'seat',
-  'correspond',
-  'linen',
-  'motif',
-  'hole',
-  'smell',
-  'smart',
-  'chaos',
-  'fuel',
-  'palace'
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
 ];
 
+function averageWordLength(array) {
+  if (array < 1) {
+    return null;
+  }
+  let sumaLongitud = 0;
+  for (i = 0; i < array.length; i++) {
+    sumaLongitud += array[i].length;
+  }
+  return sumaLongitud / array.length;
+}
 // Iteration #5: Unique arrays
 const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+  "crab",
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring"
 ];
 
+function uniquifyArray(array) {
+  let palabrasRepetidas = [];
+  let palabrasNoRepetidas = [];
+
+  for (i = 0; i < array.length; i++) {
+    if (array.indexOf(array[i], i + 1) > 0) {
+      palabrasRepetidas = array[i];
+    } else {
+      palabrasNoRepetidas = array[i];
+    }
+  }
+  return palabrasNoRepetidas;
+}
```

Comment:
Creo que en este apartado lo que queréis es añadir las palabras a un nuevo array por lo que siempre tenéis que usar el método push, si hacéis un igual cada vez que suceda va a poner un valor, no lo va a meter dentro del array